pkg/model/volumes.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2020 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/okteto/okteto/pkg/errors"
+)
+
+// HasLocalVolumes returns true if the manifest has local volumes
+func (dev *Dev) HasLocalVolumes() bool {
+	for _, v := range dev.Volumes {
+		if v.LocalPath != "" {
+			return true
+		}
+	}
+	return false
+}
+
+// HasRemoteVolumes returns true if the manifest has remote volumes
+func (dev *Dev) HasRemoteVolumes() bool {
+	for _, v := range dev.Volumes {
+		if v.LocalPath == "" {
+			return true
+		}
+	}
+	return false
+}
+
+// IsSyncFolder returns if path must be synched
+func (dev *Dev) IsSyncFolder(path string) (bool, error) {
+	found := false
+	for _, v := range dev.Volumes {
+		if v.LocalPath == "" {
+			continue
+		}
+		rel, err := filepath.Rel(v.LocalPath, path)
+		if err != nil {
+			return false, err
+		}
+		if strings.HasPrefix(rel, "..") {
+			continue
+		}
+		found = true
+		if rel != "." {
+			return false, nil
+		}
+	}
+	if found {
+		return true, nil
+	}
+	return false, errors.ErrNotFound
+}
+
+// PersistentVolumeEnabled returns true if persistent volumes are enabled for dev
+func (dev *Dev) PersistentVolumeEnabled() bool {
+	if dev.PersistentVolumeInfo == nil {
```

Comment:
can we set this when we do the SetDefault or in the deserialization for consistency?

Reply:
not related to this PR, I created an issue to handle it:
https://github.com/okteto/okteto/issues/1007